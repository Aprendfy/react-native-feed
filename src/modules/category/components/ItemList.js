import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { itemListStyles } from '../../theme/categories/styles';

export const ItemList = ({ color, title, id, onPress }) => {
  return (
    <TouchableOpacity style={[itemListStyles.itemStyle, { backgroundColor: color }]} onPress={() => onPress(title, id)}>
      <Text style={itemListStyles.textStyle}>{title}</Text>
    </TouchableOpacity>
  );
};
