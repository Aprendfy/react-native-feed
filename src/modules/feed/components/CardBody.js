import React from 'react';
import PropTypes from 'prop-types';

import { Text, View } from 'react-native';
import { CardAdsBanner } from './';
import Styles from './styles/CardBodyStyle';

const splitByBlankSpace = /\s+/;
const firstWord = 0;

const splitTextInHalf = (text, firstHalf) => {
  const words = text.split(splitByBlankSpace);
  const middleWord = words.length / 2;
  const halfWords = firstHalf ? words.slice(firstWord, middleWord) : words.slice(middleWord);

  return halfWords.join(' ');
};

export const CardBody = ({ text }) => {
  return (
    <View style={Styles.textWrapper}>
      <Text style={Styles.text}>
        {splitTextInHalf(text, true)}
      </Text>
      <CardAdsBanner />
      <Text style={Styles.text}>
        {splitTextInHalf(text)}
      </Text>
    </View>
  );
};

CardBody.propTypes = {
  text: PropTypes.string.isRequired,
};
