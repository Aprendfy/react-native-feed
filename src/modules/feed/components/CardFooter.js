import React from 'react';
import { View } from 'react-native';
import { AdMobBanner } from 'react-native-admob';
import Styles from './styles/CardFooterStyle';

export const CardFooter = () => {
  return (
    <View style={Styles.container} >
      <AdMobBanner
        bannerSize="smartBannerPortrait"
        adUnitID="ca-app-pub-5511910440808215/9505833480"
        testDeviceID="EMULATOR"
        didFailToReceiveAdWithError={error => console.log(error, 'error de banner')}
      />
    </View>
  );
};
