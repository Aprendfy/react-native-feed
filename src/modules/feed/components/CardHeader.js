import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, View, Image } from 'react-native';
import IonIcons from 'react-native-vector-icons/Ionicons';
import LinearGradient from 'react-native-linear-gradient';
import { colors } from '../../theme/styles';
import Styles from './styles/CardHeaderStyle';

export class CardHeader extends Component {
  render() {
    const { title, category, readingTime, level, color, image } = this.props;
    return (
      <View style={Styles.mainContainer}>
        <Image
          style={Styles.imageCover}
          resizeMethod="scale"
          resizeMode="cover"
          source={{ uri: image }}
        >
          <LinearGradient colors={['transparent', color]} style={Styles.firstContainer}>
            <View style={Styles.titleWrapper}>
              <Text style={Styles.title}>{title}</Text>
            </View>
          </LinearGradient>
          <View style={[Styles.secondContainer, { backgroundColor: color }]}>
            <View style={Styles.itemDetailWrapper}>
              <IonIcons name="ios-pricetag-outline" size={14} color={colors.whitePrimary} />
              <Text style={Styles.detailText}>{category}</Text>
            </View>
            <View style={Styles.itemDetailWrapper}>
              <IonIcons name="ios-time-outline" size={14} color={colors.whitePrimary} />
              <Text style={Styles.detailText}>{readingTime}</Text>
            </View>
            <View style={Styles.itemDetailWrapper}>
              <IonIcons name="ios-school" size={14} color={colors.whitePrimary} />
              <Text style={Styles.detailText}>{level}</Text>
            </View>
          </View>
        </Image>
      </View>
    );
  }

}

CardHeader.propTypes = {
  title: PropTypes.string,
  category: PropTypes.string,
  readingTime: PropTypes.string,
  level: PropTypes.string,
  color: PropTypes.string,
  image: PropTypes.string
};

CardHeader.defaultProps = {
  title: '',
  category: '',
  readingTime: '',
  level: '',
  color: '',
  image: ''
};
