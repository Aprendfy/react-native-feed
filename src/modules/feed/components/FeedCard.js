import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import { CardHeader, CardBody, CardFooter } from './';
import Styles from './styles/FeedCardStyle';

export class FeedCard extends Component {
  constructor(props) {
    super(props);
    this.state = { isCollapsed: true };
  }

  toggleIsCollapsed() {
    this.state.isCollapsed = !this.state.isCollapsed;
  }

  render() {
    const { isCollapsed } = this.state;
    const { title, category, readingTime, level, color, image, body } = this.props;
    const headerProps = { title, readingTime, level, color, image, category };

    return (
      <View style={isCollapsed ? Styles.collapsedContainer : Styles.fullContainer}>
        <CardHeader {...headerProps} />
        <CardBody text={body} />
        <CardFooter />
      </View>
    );
  }
}

FeedCard.propTypes = {
  title: PropTypes.string,
  category: PropTypes.string,
  readingTime: PropTypes.string,
  level: PropTypes.string,
  color: PropTypes.string,
  image: PropTypes.string,
  body: PropTypes.string,
};

FeedCard.defaultProps = {
  title: '',
  category: '',
  readingTime: '',
  level: '',
  color: '',
  image: '',
  body: '',
};
