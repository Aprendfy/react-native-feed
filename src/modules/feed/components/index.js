export { CardBody } from './CardBody';
export { CardAdsBanner } from './CardAdsBanner';
export { CardHeader } from './CardHeader';
export { CardSeparator } from './CardSeparator';
export { FeedCard } from './FeedCard';
export { FeedList } from './FeedList';
