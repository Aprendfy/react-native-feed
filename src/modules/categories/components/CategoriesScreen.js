import React, { Component } from 'react';
import {
  FlatList,
  View
} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Router } from '../../../router/routes';
import IonIcons from 'react-native-vector-icons/Ionicons'
import ItemList from '../../../components/categories/ItemList';
import { categoriesScreenStyles as styles } from '../../../assets/styles/categories/styles';
import { colors } from '../../../assets/styles/styles';

import * as actions from './../actions/index';

class CategoriesScreen extends Component {
  constructor(props) {
    super(props);
    this.onPressCategorie = this.onPressCategorie.bind(this);
    this.renderItem = this.renderItem.bind(this);
  }

  static route = {
    navigationBar: {
      visible: true,
      title: 'Etiquetas',
      renderRight: () => {
        return (
          <View style={styles.navButtonsContainer}>
            <View style={styles.navButton}>
              <IonIcons name="ios-pricetag-outline" size={24} color={colors.whitePrimary} />
            </View>
          </View>
        );
      }
    }

  }

  componentWillMount() {
    const { categoriesState, categoriesActions } = this.props;
    categoriesActions.fetchCategories();
  }

  renderItem({ item, index }) {
    return <ItemList title={item.title} color={item.color} id={index} onPress={this.onPressCategorie} />;
  }

  onPressCategorie(title, tab) {
    this.props.navigator.push(Router.getRoute('feed', { title, tab }));
  }

  render() {
    const { categoriesState, categoriesActions } = this.props;
    return (
      <View>
        <FlatList
          renderItem={this.renderItem}
          data={categoriesState.categories}
          keyExtractor={(item, index) => index}
        />
      </View>
    );
  }
}

export default connect(
  state => ({
    categoriesState: state.categories
  }),
  dispatch => ({
    categoriesActions: bindActionCreators(actions, dispatch),
  })
)(CategoriesScreen);
