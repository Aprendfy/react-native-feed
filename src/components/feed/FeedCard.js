import React, { Component, PropTypes } from 'react';
import { Text, View } from 'react-native';

import { AdMobBanner } from 'react-native-admob';

import { colors } from '../../assets/styles/styles';
import CardHeader from './CardHeader';

class FeedCard extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { title, category, readingTime, level, color, image, body, dateCreated } = this.props;
    return (
      <View style={{ flex: 1 }}>
        <View style={{ flex: 2 }}>
          <CardHeader
            title={title}
            tag={category}
            time={readingTime}
            level={level}
            color={color}
            image={{ uri: image }}
          />
        </View>
        <View style={{ flex: 3 }}>
          <Text style={{ color: colors.blackPrimary, fontSize: 14, padding: 5 }}>
            {body}
          </Text>
        </View>
        <View style={{ position: 'absolute', left: 0, right: 0, bottom: 0 }}>
          <AdMobBanner
            bannerSize="smartBannerPortrait"
            adUnitID="ca-app-pub-8356555649836141/9541656259"
            testDeviceID="EMULATOR"
            didFailToReceiveAdWithError={this.bannerError}
          />
        </View>
      </View>
    );
  }
}

FeedCard.propTypes = {
  title: PropTypes.string,
  category: PropTypes.string,
  readingTime: PropTypes.string,
  level: PropTypes.string,
  color: PropTypes.string,
  image: PropTypes.string,
  body: PropTypes.string,
  dateCreated: PropTypes.string,

};

FeedCard.defaultProps = {
  title: '',
  category: '',
  readingTime: '',
  level: '',
  color: '',
  image: '',
  body: '',
  dateCreated: '',

};

export default FeedCard;
