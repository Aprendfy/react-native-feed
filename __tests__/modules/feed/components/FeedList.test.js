import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { FeedCard } from '../../../../src/modules/categories/components/FeedCard';

const props = {
};

describe('<FeedCard />', () => {
  it('Should have a snapshot', () => {
    const wrapper = shallow(<FeedCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
