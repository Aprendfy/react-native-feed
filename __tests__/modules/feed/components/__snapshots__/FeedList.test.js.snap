// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedList /> Should have a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "flex": 1,
          }
    }
    tabLabel=""
>
    <FlatList
        ItemSeparatorComponent={[Function]}
        data={Array []}
        disableVirtualization={false}
        getItem={undefined}
        getItemCount={undefined}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReached={[Function]}
        onEndReachedThreshold={2}
        onEndThreshold={10}
        removeClippedSubviews={true}
        renderItem={[Function]}
        renderScrollComponent={[Function]}
        updateCellsBatchingPeriod={50}
        windowSize={21}
    />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "flex": 1,
                }
      }
      tabLabel=""
>
      <FlatList
            ItemSeparatorComponent={[Function]}
            data={Array []}
            disableVirtualization={false}
            getItem={undefined}
            getItemCount={undefined}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReached={[Function]}
            onEndReachedThreshold={2}
            onEndThreshold={10}
            removeClippedSubviews={true}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            updateCellsBatchingPeriod={50}
            windowSize={21}
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <FeedList
        color="red"
        list={Array []}
        onEndReached={[Function]}
        route={
                Object {
                        "params": Object {
                          "tab": "tab",
                          "title": "title",
                        },
                      }
        }
        tabLabel=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": FeedList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "color": "red",
          "list": Array [],
          "onEndReached": [Function],
          "route": Object {
            "params": Object {
              "tab": "tab",
              "title": "title",
            },
          },
          "tabLabel": "",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
          tabLabel=""
>
          <FlatList
                    ItemSeparatorComponent={[Function]}
                    data={Array []}
                    disableVirtualization={false}
                    getItem={undefined}
                    getItemCount={undefined}
                    horizontal={false}
                    initialNumToRender={10}
                    keyExtractor={[Function]}
                    maxToRenderPerBatch={10}
                    numColumns={1}
                    onEndReached={[Function]}
                    onEndReachedThreshold={2}
                    onEndThreshold={10}
                    removeClippedSubviews={true}
                    renderItem={[Function]}
                    renderScrollComponent={[Function]}
                    updateCellsBatchingPeriod={50}
                    windowSize={21}
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
          tabLabel=""
>
          <FlatList
                    ItemSeparatorComponent={[Function]}
                    data={Array []}
                    disableVirtualization={false}
                    getItem={undefined}
                    getItemCount={undefined}
                    horizontal={false}
                    initialNumToRender={10}
                    keyExtractor={[Function]}
                    maxToRenderPerBatch={10}
                    numColumns={1}
                    onEndReached={[Function]}
                    onEndReachedThreshold={2}
                    onEndThreshold={10}
                    removeClippedSubviews={true}
                    renderItem={[Function]}
                    renderScrollComponent={[Function]}
                    updateCellsBatchingPeriod={50}
                    windowSize={21}
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FeedList
    color="red"
    list={Array []}
    onEndReached={[Function]}
    route={
        Object {
            "params": Object {
              "tab": "tab",
              "title": "title",
            },
          }
    }
    tabLabel=""
/>,
}
`;
